### PR TITLE
Daily DMP Data Fetching and Updating Workflow

### PR DESCRIPTION
**Pull request by ao508**
Signed-off-by: Angelica Ochoa <15623749+ao508@users.noreply.github.com>

### CODE REVIEW

**Comment by inodb** [comment]
@ao508 this is really helpful! Thanks for doing this!

**Comment by inodb** [comment]
@angelica Not sure if it's helpful for me to review (yet) but just a few minor notes as someone not familiar with the process:

- github state: what is that and how is it used?
- dmp updates workflow seems to contain a lot of bash code. How do we make sure the documentation doesn’t get out of sync with the actual scripts? I assume there is some master script that runs all this?

**Comment by sheridancbio** [comment]
There are extensive blocks of code shown in the documentation. It seems to me that these will soon be out of date unless we keep maintaining the documentation as we make code updates.

@ao508 suggests that we will automate a process to apply the headings and formatting to the code in order to do automatic updating of this documentation. It seems to me that that it will be a challenge to develop/maintain such automated code update scripts too.

File: import-scripts/dmp-import-vars-functions.sh [hunk 1]
```diff
@@ -319,7 +319,7 @@ function remove_raw_clinical_timeline_data_files {
     done
     # remove raw timeline files except cbio format timeline file
     for f in $STUDY_DIRECTORY/data_timeline*; do
-        if [ $f != *"data_timeline.txt"* ] ; then
+        if [[ $f != *"data_timeline.txt"* ]] ; then
```
Comment (sheridancbio) [inline import-scripts/dmp-import-vars-functions.sh]: I don't understand this line ... the fix seems fine, but I don't get the string comparison to a de-globbed (wildcard expanded) list of filenames. But that is a pre-existing bug (in my opinion)

**Comment by ao508** [comment]
@inodb @sheridancbio Future plans include making a script that will generate a formatted workflow document. As per our morning discussion, this doc is good for merge now as is given the future plans.

**Comment by inodb** [comment]
Thanks @ao508 ! What about the github states? Does that mean step 1/2/3 or do we keep track of the state id somewhere in code/config? Maybe i missed it somewhere in the docs

Re docs generation: could work. Is the idea to parse comments from some bash script? It might also work to split up the files a bit more so you can e.g. add a link to a script in the docs which has more detailed comments at the top of the file. As a more general comment: there are ofc many tools to generate docs from code, but if we have to develop our own tooling it could also be indicative of the code becoming too complex and that we've outgrown the existing setup. There are e.g. workflow languages that were specifically developed to handle these complexities, e.g. nextflow provides workflow tracing out of the box: https://www.nextflow.io/docs/latest/tracing.html

**Comment by ao508** [comment]
@inodb Thanks for the pointer! 

Yes, we plan to capture GitHub states as well. The formatting will not be as "pretty" but I'm sure whatever automated workflow doc solution we end up using will be sufficient for what we need.

**Pull request merged**